ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/mac-security.j2

```diff
@@ -20,7 +20,7 @@ mac security
       cipher {{ profile.cipher }}
 {%         endif %}
 {%         for connection_key in profile.connection_keys | arista.avd.natural_sort('id') %}
-{%             if connection_key.encrypted_key is arista.avd.defined %}
+{%             if connection_key.encrypted_key is arista.avd.defined and connection_key.id is arista.avd.defined %}
```

Comment:
```suggestion
{%             if connection_key.encrypted_key is arista.avd.defined %}
```
not required since arista.avd.natural_sort('id') is checking it.